modules/RTC/ScreenObtainer.js

```diff
@@ -76,11 +76,9 @@ const ScreenObtainer = {
         this.options = options = options || {};
         gumFunction = gum;
 
-        if (RTCBrowserType.isFirefox()) {
-            initFirefoxExtensionDetection(options);
-        }
-
-        if (RTCBrowserType.isNWJS()) {
+        if (this.options.disableDesktopSharing) {
```

Comment:
I was actually thinking of having something like:

if (this.options.disableDesktopSharing) {
logger.info('Desktop sharing disabled');
this.obtainStream = null;
return;
}

before all the ifs starting with if (RTCBrowserType.isNWJS()) ...